pkg/util/kubernetes/kubelet/podwatcher.go

```diff
@@ -122,3 +148,23 @@ func (w *PodWatcher) Expire() ([]string, error) {
 func (w *PodWatcher) GetPodForEntityID(entityID string) (*Pod, error) {
 	return w.kubeUtil.GetPodForEntityID(entityID)
 }
+
+// digestMap return a unique hash for a map
+// used to track changes in labels and annotations values
+func digestMap(m map[string]string) string {
```

Comment:
I think this could take both annotations and labels, or even the whole metadata object, and return a single string for the pod (and it could be named `digestPodMeta` in this case). That's just a proposal, might not be the best, but the name digestMap confused me at first because I expected it to digest the whole map and it actually hashes values only.

Reply:
what I suggest:
- rename the function to `digestMapValues`
- add another function `digestPodMeta` which takes a metadata object as an argument
- `digestPodMeta` calls `digestMapValues` twice to calculate digests for labels and annotations
- `digestPodMeta` returns a single pod digest from both labels and annotations digests

I'd keep the `digestMapValues` logic (keys sorting and map values hash calculation) in a separate function to avoid duplicating it for 2 maps (labels and annotations) in `digestPodMeta`.
WDYT?

Reply:
That sounds great 👍 let's just make sure to define explicitly in the comment of the `digestPodMeta` in which order we write annotation and label hashes.

Reply:
done 👍